src/storage/resourceLoader.ts

```diff
@@ -196,7 +201,9 @@ export class CCloudResourceLoader extends ResourceLoader {
    */
   private schemaRegistryCacheStates: Map<string, boolean | Promise<void>> = new Map();
 
-  private constructor() {
+  // Singleton class. Use getInstance() to get the singleton instance.
+  // (Only public for testing / signon mocking purposes.)
+  constructor() {
```

Comment:
We can still keep this `private`, but would just need to set `CCloudResourceLoader["instance"] = null` in a test suite's `afterEach()` hook

Reply:
No, alas. setting it to private, it then doesn't fulfill Constructor interface required by the signon mocking when combined by parameterizing the tests:

<img width="748" alt="image" src="https://github.com/user-attachments/assets/2222e7f0-7627-4b47-9168-210d7a47137b">